File: app/src/main/java/com/example/projectappdoctruyen/Activity/ReadStoryActivity.java
package com.example.projectappdoctruyen.Activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.FragmentStatePagerAdapter;
import androidx.viewpager.widget.ViewPager;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.WindowManager;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.example.projectappdoctruyen.Adapter.ViewPagerReadStoryAdapter;
import com.example.projectappdoctruyen.Database.DatabaseHelper;
import com.example.projectappdoctruyen.Model.Chapter;
import com.example.projectappdoctruyen.Model.Story;
import com.example.projectappdoctruyen.R;
import com.example.projectappdoctruyen.SettingSharedPreferences.SharedPreferences_Utils;
import com.google.android.gms.ads.AdView;

import java.util.ArrayList;
import java.util.List;

public class ReadStoryActivity extends AppCompatActivity {
    public  Toolbar testToolBarId;
    public  LinearLayout linearLayout_Visible;
    private Chapter chapter;
    private AdView adView;
    private SharedPreferences_Utils sharedPreferencesUtils;
    private ViewPager viewPager;
    private ViewPagerReadStoryAdapter viewPagerReadStoryAdapter;
    private List<Chapter> chapterArrayList;
    private DatabaseHelper databaseHelper;
    private Story story;
        private ArrayList<Story> storyArrayList;
    private TextView txtBack ,txtCurrentNumber, txtTotalNumber, txtNext;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,WindowManager.LayoutParams.FLAG_FULLSCREEN);
        setContentView(R.layout.activity_read_story);
        sharedPreferencesUtils = new SharedPreferences_Utils(this);
        databaseHelper = new DatabaseHelper(this);
        init();
        actionToolBar();
        dataIntent();
        chapterArrayList =  databaseHelper.getChapTer(String.valueOf(chapter.getIdStory()));
        int idStory = chapter.getIdStory();
        story = databaseHelper.getStoryById(idStory+"");
        setAdapterViewPager();
        int  index = indexChapter(chapterArrayList,chapter.getIdChapter());
        txtCurrentNumber.setText(index+ 1 +"");
        txtTotalNumber.setText(String.valueOf(chapterArrayList.size()));
        viewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
            }

            @Override
            public void onPageSelected(int position) {
                txtCurrentNumber.setText(String.valueOf(position + 1));
                if (position ==0){
                    txtBack.setVisibility(View.GONE);
                    txtNext.setVisibility(View.VISIBLE);
                }else if(position == chapterArrayList.size() -1 ){
                    txtBack.setVisibility(View.VISIBLE);
                    txtNext.setVisibility(View.GONE);
                }else {
                    txtNext.setVisibility(View.VISIBLE);
                    txtBack.setVisibility(View.VISIBLE);
                }
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });
        txtBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                viewPager.setCurrentItem(viewPager.getCurrentItem() -1);
            }
        });
        txtNext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                viewPager.setCurrentItem(viewPager.getCurrentItem()+1);
            }
        });
    }
    private  int indexChapter(List<Chapter> list, int id){
        for (int i = 0;i<list.size();i++){
            if (list.get(i).getIdChapter() == id){
                return i;
            }
        }return -1;
    }
    private void setAdapterViewPager() {
        viewPagerReadStoryAdapter = new ViewPagerReadStoryAdapter(getSupportFragmentManager(),
                FragmentStatePagerAdapter.BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT,chapterArrayList);
        viewPager.setAdapter(viewPagerReadStoryAdapter);
        int indexofchapter  = -1;
        for (int i = 0;i<chapterArrayList.size();i++){
            if (chapterArrayList.get(i).getIdChapter() == chapter.getIdChapter()){
                indexofchapter = i;
            }
        }
        viewPager.setCurrentItem(indexofchapter);
    }
    private void dataIntent() {
        Intent intent = getIntent();
        if (intent != null){
            chapter = (Chapter) intent.getSerializableExtra("Chapter");
            Log.d("FFASDAS",chapter.getTitle()+"");
        }
    }
    private void actionToolBar() {
        testToolBarId = findViewById(R.id.testToolBarId);
        setSupportActionBar(testToolBarId);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        testToolBarId.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }
    private void init() {
        txtBack = findViewById(R.id.txtBack);
        txtCurrentNumber = findViewById(R.id.txtCurrentNumber);
        txtTotalNumber = findViewById(R.id.txtTotalNumber);
        txtNext = findViewById(R.id.txtNext);
        adView = findViewById(R.id.adView);
        viewPager = findViewById(R.id.fr_container_test);
        linearLayout_Visible = findViewById(R.id.linearLayout_Visible);


    }
}

File: app/src/main/java/com/example/projectappdoctruyen/Activity/ReadStoryActivity.kt
package com.example.projectappdoctruyen.Activity

import android.os.Bundle
import android.view.View
import android.view.WindowManager
import android.widget.LinearLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.FragmentStatePagerAdapter
import androidx.viewpager.widget.ViewPager
import androidx.viewpager.widget.ViewPager.OnPageChangeListener
import com.example.projectappdoctruyen.Adapter.ViewPagerReadStoryAdapter
import com.example.projectappdoctruyen.Database.DatabaseHelper
import com.example.projectappdoctruyen.Model.Chapter
import com.example.projectappdoctruyen.Model.Story
import com.example.projectappdoctruyen.R
import com.example.projectappdoctruyen.SettingSharedPreferences.SharedPreferences_Utils
import com.google.android.gms.ads.AdView
import java.util.*

class ReadStoryActivity : AppCompatActivity() {
    var testToolBarId: Toolbar? = null
    var linearLayout_Visible: LinearLayout? = null
    private var chapter: Chapter? = null
    private var adView: AdView? = null
    private var sharedPreferencesUtils: SharedPreferences_Utils? = null
    private var viewPager: ViewPager? = null
    private var viewPagerReadStoryAdapter: ViewPagerReadStoryAdapter? = null
    private var chapterArrayList: List<Chapter?>? = null
    private var databaseHelper: DatabaseHelper? = null
    private var story: Story? = null
    private val storyArrayList: ArrayList<Story>? = null
    private var txtBack: TextView? = null
    private var txtCurrentNumber: TextView? = null
    private var txtTotalNumber: TextView? = null
    private var txtNext: TextView? = null
    lateinit var linearBottom : LinearLayout
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)
        setContentView(R.layout.activity_read_story)
        sharedPreferencesUtils = SharedPreferences_Utils(this)
        databaseHelper = DatabaseHelper(this)
        init()
        actionToolBar()
        dataIntent()
        chapterArrayList = databaseHelper!!.getChapTer(chapter!!.idStory.toString())
        val idStory = chapter!!.idStory
        story = databaseHelper!!.getStoryById(idStory.toString() + "")
        setAdapterViewPager()
        val index = indexChapter(chapterArrayList, chapter!!.idChapter!!)
        val indexsumone = index+1
        txtCurrentNumber!!.text = indexsumone.toString()

        txtTotalNumber!!.text = chapterArrayList!!.size.toString()
        viewPager!!.addOnPageChangeListener(object : OnPageChangeListener {
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {}
            override fun onPageSelected(position: Int) {
                txtCurrentNumber!!.text = (position + 1).toString()
                if (position == 0) {
                    txtBack!!.visibility = View.GONE
                    txtNext!!.visibility = View.VISIBLE
                } else if (position == chapterArrayList!!.size - 1) {
                    txtBack!!.visibility = View.VISIBLE
                    txtNext!!.visibility = View.GONE
                } else {
                    txtNext!!.visibility = View.VISIBLE
                    txtBack!!.visibility = View.VISIBLE
                }
            }

            override fun onPageScrollStateChanged(state: Int) {}
        })
        txtBack!!.setOnClickListener { viewPager!!.currentItem = viewPager!!.currentItem - 1 }
        txtNext!!.setOnClickListener { viewPager!!.currentItem = viewPager!!.currentItem + 1 }

        setOnclickList()
    }
    fun getViewPager(): ViewPager? {
        if (null == viewPager) {
            viewPager = findViewById<ViewPager>(R.id.fr_container_test)
        }
        return viewPager
    }
    private fun setOnclickList() {
        linearBottom.setOnClickListener {

//                showDialogNotification();
            val bottomSheet = ExampleBottomSheetDialog()
            bottomSheet.show(supportFragmentManager, "exampleBottomSheet")
        }
    }

    private fun indexChapter(list: List<Chapter?>?, id: Int): Int {
        for (i in list!!.indices) {
            if (list[i]!!.idChapter == id) {
                return i
            }
        }
        return -1
    }

    private fun setAdapterViewPager() {
        viewPagerReadStoryAdapter = ViewPagerReadStoryAdapter(supportFragmentManager,
                FragmentStatePagerAdapter.BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT, chapterArrayList)
        viewPager!!.adapter = viewPagerReadStoryAdapter
        var indexofchapter = -1
        for (i in chapterArrayList!!.indices) {
            if (chapterArrayList!![i]!!.idChapter == chapter!!.idChapter) {
                indexofchapter = i
            }
        }
        viewPager!!.currentItem = indexofchapter
    }

    private fun dataIntent() {
        val intent = intent
        if (intent != null) {
            chapter = intent.getSerializableExtra("Chapter") as Chapter?
        }
    }

    private fun actionToolBar() {
        testToolBarId = findViewById(R.id.testToolBarId)
        setSupportActionBar(testToolBarId)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        testToolBarId!!.setNavigationOnClickListener(View.OnClickListener { finish() })
    }

    private fun init() {
        viewPager = findViewById<ViewPager>(R.id.fr_container_test)
        linearBottom = findViewById(R.id.linearBottom)
        txtBack = findViewById(R.id.txtBack)
        txtCurrentNumber = findViewById(R.id.txtCurrentNumber)
        txtTotalNumber = findViewById(R.id.txtTotalNumber)
        txtNext = findViewById(R.id.txtNext)
        adView = findViewById(R.id.adView)
        linearLayout_Visible = findViewById(R.id.linearLayout_Visible)
    }
}